app/ports/management/commands/load-rsync.py

```diff
@@ -0,0 +1,29 @@
+from django.core.management.base import BaseCommand, CommandError
+
+from ports.models import Port, LastPortIndexUpdate
+from parsing_scripts import rsync_operations
+
+
+class Command(BaseCommand):
+
+    help = "Populates the database with Initial data after fetching portindex.json from rsync"
+
+    def handle(self, *args, **options):
+        if not Port.objects.count() > 0:
+            # Fetch from rsync
+            rsync_operations.sync()
+
+            # Open the file
+            data = rsync_operations.open_file()
+
+            # Start populating
+            Port.load(data['ports'])
+
+            # Write the commit hash into database
+            last_commit = LastPortIndexUpdate.objects.all().first()
+            last_commit.git_commit_hash = data['info']['commit']
+            last_commit.save()
```

Comment:
While you can leave this as it is for a little while, this field should not be updated unless database population is known to be successful.

What would make slightly more sense to me would be to pass the full json to `Port.load` and then the commit would be updated at the end of that function rather than here. @umeshksingla?

Reply:
Thank you, but if `Port.load()` throws an error, the command will not proceed either.